x/gov/router.go

```diff
@@ -0,0 +1,46 @@
+package gov
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/cosmos/cosmos-sdk/x/gov/proposal"
+)
+
+// Router is a map from string to proposal.Handler
+// copied and modified from baseapp/router.go
+type Router interface {
+	AddRoute(r string, h proposal.Handler) (rtr Router)
+	Route(path string) (h proposal.Handler)
+}
+
+type router struct {
+	routes map[string]proposal.Handler
+}
+
+var _ Router = (*router)(nil)
+
+// Constructs new router
+func NewRouter() *router {
+	return &router{
+		routes: make(map[string]proposal.Handler),
+	}
+}
+
+var isAlphaNumeric = regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString
+
+func (rtr *router) AddRoute(path string, h proposal.Handler) Router {
+	if !isAlphaNumeric(path) {
+		panic("route expressions can only contain alphanumeric characters")
```

Comment:
why panic ? this should return an error

Reply:
`AddRoute` is (supposed to be) only called on the initialization time. Any initialization time error should panic in order to prevent starting an invalid chain. Same behavior with baseapp.router.